cli/cmd/upgrade.go

```diff
@@ -379,6 +385,32 @@ func fetchTLSSecret(k kubernetes.Interface, webhook string, options *upgradeOpti
 	return value, nil
 }
 
+func ensureIssuerCertWorksWithAllProxies(k kubernetes.Interface, cred *tls.Cred) error {
+	meshedPods, err := healthcheck.GetMeshedPodsIdentitiyData(k, "")
+	var problematicPods []string
+	if err != nil {
+		return err
+	}
+	for _, pod := range meshedPods {
+		roots, err := tls.DecodePEMCertPool(pod.Anchors)
+
+		if roots != nil {
+			err = cred.Verify(roots, "")
+		}
+
+		if err != nil {
+			problematicPods = append(problematicPods, fmt.Sprintf("* %s", pod.Name))
```

Comment:
I think we need the pod namespace in here too, like in https://github.com/linkerd/linkerd2/pull/3821/files#diff-d4056ff163bcf2aeacefb2a34164563cR1445